motionplan/motionPlanner.go

```diff
@@ -385,3 +382,90 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
```

Comment:
Nick S. brought this up yesterday when we were talking.
Do we think this function should be renamed? e.g. `CheckPlanForCollision` to be more explicit about what the function does.
lmk!

Reply:
I think there's the potential for this function to be expanded to interact with more constraints in the future.

For example, as soon as we allow the specification of "don't diverge from the straight-line path between these points", or "stay on that road".

So I would keep the name.